Lesson6/src/model/Person.java

```diff
@@ -0,0 +1,130 @@
+package model;
+
+import exceptions.NoAddressException;
+import exceptions.CantSpeakException;
+import exceptions.SurNameException;
+import interfaces.IPerson;
+import interfaces.Speaker;
+
+
+public class Person implements IPerson, Speaker {
+    private String sex;
+    private int age;
+    private String name;
+    private String surname;
+    private int height;
+    private Address address;
+
+    public Person(String sex, Integer age, String name, String surname, Integer height) {
+        this.sex = sex;
+        this.age = age;
+        setName(name);
+        setSurname(surname);
+        this.height = height;
+    }
+
+    private String personSurNameFilter(String surName) throws SurNameException {
+        if (surName == null || surName.isEmpty()) {
+            throw new SurNameException("You sent 'null'.");
+        }
+        String text = surName.trim().toLowerCase();
+        String textCapitalLetter = text.substring(0, 1).toUpperCase();
+        String textSmallLetters = text.substring(1).toLowerCase();
+        String textCommon = textCapitalLetter + textSmallLetters;
+        if (textCommon.indexOf(" ") > 0) {
+            try {
+                throw new SurNameException("Wrong input " + textCommon + ". Don't use 'space'.");
+            } catch (SurNameException e) {
```

Comment:
you don't want to catch it in the constructor, let code that creates person handle it

Reply:
I did this in the Main class, when I created a person. However, when the exception  was caught the for loop was broke and person wasn't created.